tests/TopDownProteomics.Tests/ProteoformClassifierTest.cs

```diff
@@ -0,0 +1,59 @@
+﻿using NUnit.Framework;
+using System.Collections.Generic;
+using System.Linq;
+using TopDownProteomics.ProForma;
+using TopDownProteomics.Proteomics;
+
+namespace TopDownProteomics.Tests
+{
+    [TestFixture]
+    public static class ProteoformClassifierTest
+    {
+        [Test]
+        [TestCase("EMEVEESPEK", 1, "1")]
+        [TestCase("EM[UNIMOD:35]EVEES[UNIMOD:21]PEK", 1, "1")]
+        [TestCase("[+42]-EMEVEES[UNIMOD:21]PEK", 1, "2B")]
+        [TestCase("EM[UNIMOD:35]EVEESPEK-[+42]", 1, "2B")]
+        [TestCase("{Phospho}EM[UNIMOD:35]EVEESPEK", 1, "2A")]
+        [TestCase("{+80}EM[UNIMOD:35]EVEESPEK", 1, "3")]
+        [TestCase("SEQUEN[Formula:C12H20O2]CE", 1, "1")]
+        [TestCase("[Phospho]?EMEVTSESPEK", 1, "2A")]
+        [TestCase("EMEVT[#g1]S[#g1]ES[Phospho#g1]PEK", 1, "2A")]
+        [TestCase("EM[+15.9949]EVEES[-79.9663]PEK", 1, "2B")]
+        [TestCase("EMEVEESPEK[Acetyl|Trimethyl]", 1, "2B")] // RTF: Ask into this!
+        [TestCase("EMEVEE(?SP)EK", 1, "2C")]
+        [TestCase("PROTEOSFORMSISK(?N)", 1, "2C")]
+        [TestCase("EMEVEESPEKB", 1, "2C")]
+        [TestCase("EMEVEESPEKJ", 1, "2C")]
+        [TestCase("EMEVEESPEKX", 1, "2C")]
+        [TestCase("EMEVEESPEKZ", 1, "2C")]
+        [TestCase("EMEVEESPEK", 2, "2D")]
+        [TestCase("PROT(EOSFORMS)[+19.0523]ISKN", 1, "3")]
+        [TestCase("PROT(EOSFORMS)[+19.0523]ISK(?N)", 1, "4")]
```

Comment:
Doesn't this denote ambiguous sequence order? If there's only one amino acid, this ordering only has one possibility, and thus this one should return `3`.

Reply:
I believe the thought in this specific case is that the sequence can be 'N' or 'GG'. So, the classification is blindly assuming that if someone annotated sequence ambiguity, they meant it. You are right that, in some cases, we could logic away the ambiguity, but that would be hard.